infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -292,13 +298,24 @@ private void rangeWrite(CorfuPayloadMsg<RangeWriteMsg> msg,
     }
 
     /**
-     * Resets the log unit server.
+     * Resets the log unit server via the BatchWriter.
      * Warning: Clears all data.
      */
     @ServerHandler(type = CorfuMsgType.RESET_LOGUNIT)
-    private void resetLogUnit(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        streamLog.reset();
-        dataCache.invalidateAll();
+    private synchronized void resetLogUnit(CorfuPayloadMsg<Long> msg,
+                                           ChannelHandlerContext ctx, IServerRouter r) {
+        // Check if the reset request is with an epoch greater than the last reset epoch seen.
+        // and should be equal to the current router epoch to prevent stale reset requests from
+        // wiping out the data.
+        if (msg.getPayload() > serverContext.getLogUnitResetEpoch()
+                && msg.getPayload() == serverContext.getServerEpoch()) {
+            batchWriter.reset();
```

Comment:
Why make the reset call ignore the epoch just to check it again?
You set the new epoch, then call reset with the same epoch. You just need to check 
msg.getPayload() > serverContext.getLogUnitResetEpoch()

Reply:
So this is different compared to the router epoch check.
If you see the LayoutManagementView Class, this is performed after a seal. The layout slot has not been filled yet. The epoch in the message payload is the sealed epoch (no layout committed yet).

This is required to prevent stale resets from an older epoch. If the log unit has never seen any resets, its reset epoch will be set to -1. It will allow any reset.

Reply:
Are you sure?

I understand the greater than logunit epoch check. I'm pointing out that setServerEpoch (after the seal) will store the epoch in the same key that validateEpoch uses, so subsequent calls will check if that msg.getPayload equals serverContext.getServerEpoch, so what is the point of setting ignore epoch, but then making the same check again in the handler ?